### PR TITLE
[EPMEDU-981] replace 4 digit code with 6

### PR DESCRIPTION
**Pull request by aklepets**

### CODE REVIEW

File: feature/signupflow/entercode/src/main/java/com/epmedu/animeal/signup/entercode/presentation/ui/ZeroContentPaddingOutlinedTextField.kt [hunk 1]
```diff
@@ -0,0 +1,105 @@
+package com.epmedu.animeal.signup.entercode.presentation.ui
```
Comment (Fileek) [inline feature/signupflow/entercode/src/main/java/com/epmedu/animeal/signup/entercode/presentation/ui/ZeroContentPaddingOutlinedTextField.kt]: Move it please to the foundation module into the input folder

File: feature/signupflow/entercode/src/main/java/com/epmedu/animeal/signup/entercode/presentation/ui/ZeroContentPaddingOutlinedTextField.kt [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+                    placeholder = placeholder,
+                    label = label,
+                    leadingIcon = leadingIcon,
+                    trailingIcon = trailingIcon,
+                    singleLine = singleLine,
+                    enabled = enabled,
+                    isError = isError,
+                    interactionSource = interactionSource,
+                    colors = colors,
+                    contentPadding = PaddingValues(0.dp),
```
Comment (Andrii-Chornyi) [inline feature/signupflow/entercode/src/main/java/com/epmedu/animeal/signup/entercode/presentation/ui/ZeroContentPaddingOutlinedTextField.kt]: I would suggest not removing paddings altogether but just decreasing it, from 16.dp to, let's say, 4.dp, or even rename this compose element to something like "CustomContentPaddingOutlinedTextField" and to pass content padding as a parameter.

File: feature/signupflow/entercode/src/main/java/com/epmedu/animeal/signup/entercode/presentation/ui/CodeRow.kt [hunk 3]
```diff
@@ -50,14 +50,14 @@ private fun CodeRowPreview() {
     AnimealTheme {
         Column {
             CodeRow(
-                code = listOf(1, 2, null, null),
+                code = listOf(1, 2, null, null, null, null),
```
Comment (Fileek) [inline feature/signupflow/entercode/src/main/java/com/epmedu/animeal/signup/entercode/presentation/ui/CodeRow.kt:53]: Can you also replace list with nulls in EnterCodeScreenPreview with emptyCode() call from companion in viewmodel?

File: feature/signupflow/entercode/src/main/java/com/epmedu/animeal/signup/entercode/presentation/ui/ZeroContentPaddingOutlinedTextField.kt (hunk 2 above)
Comment (Fileek) [reply feature/signupflow/entercode/src/main/java/com/epmedu/animeal/signup/entercode/presentation/ui/ZeroContentPaddingOutlinedTextField.kt]: I agree with passing content padding as a parameter

File: feature/signupflow/entercode/src/main/java/com/epmedu/animeal/signup/entercode/presentation/ui/ZeroContentPaddingOutlinedTextField.kt [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+    colors: TextFieldColors = TextFieldDefaults.outlinedTextFieldColors()
+) {
+    // If color is not provided via the text style, use content color as a default
+    val textColor = textStyle.color.takeOrElse {
+        colors.textColor(enabled).value
+    }
+    val mergedTextStyle = textStyle.merge(TextStyle(color = textColor))
+
+    @OptIn(ExperimentalMaterialApi::class)
+    (
```
Comment (Fileek) [inline feature/signupflow/entercode/src/main/java/com/epmedu/animeal/signup/entercode/presentation/ui/ZeroContentPaddingOutlinedTextField.kt]: These parentheses are definitely redundant

File: feature/signupflow/entercode/src/main/java/com/epmedu/animeal/signup/entercode/presentation/ui/ZeroContentPaddingOutlinedTextField.kt [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Shape
+import androidx.compose.ui.graphics.SolidColor
+import androidx.compose.ui.graphics.takeOrElse
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.input.VisualTransformation
+import androidx.compose.ui.unit.dp
+
+@Composable
```
Comment (Fileek) [inline feature/signupflow/entercode/src/main/java/com/epmedu/animeal/signup/entercode/presentation/ui/ZeroContentPaddingOutlinedTextField.kt]: Add @Suppress("LongParameterList") annotation for detekt

**Review by Fileek — APPROVED** [review]

**Review by carlossantiagomenendez — APPROVED** [review]

**Pull request merged**